jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java

```diff
@@ -545,6 +562,60 @@ public boolean handle()
         return !suspended;
     }
 
+    public void ensureConsumeAllOrNotPersistent()
+    {
+        switch (_request.getHttpVersion())
+        {
+            case HTTP_1_0:
+                if (_request.getHttpInput().consumeAll())
+                    return;
+
+                // Remove any keep-alive value in Connection headers
+                _response.getHttpFields().computeField(HttpHeader.CONNECTION, (h, fields) ->
+                {
+                    if (fields == null || fields.isEmpty())
+                        return null;
+                    String v = fields.stream()
+                        .flatMap(field -> Stream.of(field.getValues()).filter(s -> !HttpHeaderValue.KEEP_ALIVE.is(s)))
+                        .collect(Collectors.joining(", "));
+                    if (StringUtil.isEmpty(v))
+                        return null;
+
+                    return new HttpField(HttpHeader.CONNECTION, v);
+                });
+                break;
+
+            case HTTP_1_1:
+                if (_request.getHttpInput().consumeAll())
+                    return;
+
+                // Add close value to Connection headers
+                _response.getHttpFields().computeField(HttpHeader.CONNECTION, (h, fields) ->
+                {
+                    if (fields == null || fields.isEmpty())
+                        return HttpConnection.CONNECTION_CLOSE;
+
+                    if (fields.stream().anyMatch(f -> f.contains(HttpHeaderValue.CLOSE.asString())))
+                    {
+                        if (fields.size() == 1)
+                            return fields.get(0);
+
+                        return new HttpField(HttpHeader.CONNECTION, fields.stream()
+                            .flatMap(field -> Stream.of(field.getValues()))
+                            .collect(Collectors.joining(", ")));
+                    }
+
+                    return new HttpField(HttpHeader.CONNECTION, fields.stream()
+                        .flatMap(field -> Stream.of(field.getValues()))
+                        .collect(Collectors.joining(", ")) + ", " + HttpHeaderValue.CLOSE.asString());
```

Comment:
Since many clients also add `Connection: keep-alive` is it worth to remove it even if the protocol is HTTP/1.1?

Reply:
There's also `Connection: upgrade`?

Reply:
removing keep-alive as well makes the code rather complex.  Close should always override keep-alive.... not so in our generator for HTTP/1.0, I will change that.

Reply:
OK I removed the `keep-alive` as well and added unit tests for all the variations... good thing too as I was adding an extra ',' in some circumstances.